src/Database/DBStructure.php

```diff
@@ -1557,12 +1557,17 @@ public static function definition() {
 		$database["process"] = [
 				"comment" => "Currently running system processes",
 				"fields" => [
-						"pid" => ["type" => "int unsigned", "not null" => "1", "primary" => "1", "comment" => ""],
+						"id" => ["type" => "int unsigned", "not null" => "1", "extra" => "auto_increment", "primary" => "1", "comment" => "Global Unique ID"],
+						"pid" => ["type" => "int unsigned", "comment" => "PID of the process"],
+						"parent" => ["type" => "int unsigned", "comment" => "ID of the parent process"],
+						"type" => ["type" => "varchar(5)", "not null" => "1", "comment" => "Type of the process"],
 						"command" => ["type" => "varbinary(32)", "not null" => "1", "default" => "", "comment" => ""],
 						"created" => ["type" => "datetime", "not null" => "1", "default" => NULL_DATE, "comment" => ""],
 						],
 				"indexes" => [
-						"PRIMARY" => ["pid"],
+						"PRIMARY" => ["id"],
+						"id_parent" => ["id", "parent"],
+						"id_pid" => ["id", "pid"],
```

Comment:
What is the sense of these indexes? If "id" is unique, then adding the parent doesn't make sense.

Reply:
It should cover the most used accesspath and increase their access speed. So if I select with ID and parent, it's enough to have parent indexed?

Reply:
When "id" is unique it doesn't make sense to query for "id" and another field. But I have to have a deeper look at the code to give a better feedback.

Reply:
Business Case:
Give me the process with ID = ? and parent = same parent as the current process
if result => check PID
if not => not same node, so don't check

Reply:
Nonetheless you can simply drop the two indexes.